### PR TITLE
pre-commit: test PR77709

### PR DESCRIPTION
**Pull request by dtcxzyw**
PR Link: https://github.com/llvm/llvm-project/pull/77709

### CODE REVIEW

File: bench/qemu/optimized/target_riscv_translate.c.ll [hunk 1]
```diff
@@ -67652,14 +67672,34 @@ if.else:                                          ; preds = %land.lhs.true6, %ge
...
+    i8 1, label %call.1
+    i8 2, label %call.2
+    i8 3, label %call.3
+  ]
+
+default.switch.case.unreachable:                  ; preds = %if.else
+  unreachable
+
+call.0:                                           ; preds = %if.else
+  tail call fastcc void @gen_helper_vmv_v_x_b(ptr noundef %call18, ptr noundef %call17, ptr noundef %32, ptr noundef %call43) #13
```
[the remark this answers is not in the document]
Comment (dtcxzyw) [reply bench/qemu/optimized/target_riscv_translate.c.ll:67687]:
```
; Function Attrs: nounwind sspstrong uwtable
define internal void @gen_helper_vmv_v_x_b(ptr noundef %arg1, ptr noundef %arg2, ptr noundef %arg3, ptr noundef %arg4) #2 {
entry:
  %0 = tail call align 8 ptr @llvm.threadlocal.address.p0(ptr align 8 @tcg_ctx)
  %1 = load ptr, ptr %0, align 8
  %2 = ptrtoint ptr %arg1 to i64
  %add.ptr.i.i = getelementptr i8, ptr %1, i64 %2
  %3 = ptrtoint ptr %arg2 to i64
  %add.ptr.i.i1 = getelementptr i8, ptr %1, i64 %3
  %4 = ptrtoint ptr %arg3 to i64
  %add.ptr.i.i2 = getelementptr i8, ptr %1, i64 %4
  %5 = ptrtoint ptr %arg4 to i64
  %add.ptr.i = getelementptr i8, ptr %1, i64 %5
  tail call void @tcg_gen_call4(ptr noundef nonnull @helper_info_vmv_v_x_b, ptr noundef null, ptr noundef %add.ptr.i.i, ptr noundef %add.ptr.i.i1, ptr noundef %add.ptr.i.i2, ptr noundef %add.ptr.i) #13
  ret void
}

```

[the remark this answers is not in the document]
Comment (nikic) [reply bench/qemu/optimized/target_riscv_translate.c.ll:67687]: Ah, maybe it's due to the self-reference passed to tcg_gen_call4?

[the remark this answers is not in the document]
Comment (dtcxzyw) [reply bench/qemu/optimized/target_riscv_translate.c.ll:67687]: > Ah, maybe it's due to the self-reference passed to tcg_gen_call4?

No, the pointer passed to `tcg_gen_call4` is `helper_info_vmv_v_x_b`.

[the remark this answers is not in the document]
Comment (nikic) [reply bench/qemu/optimized/target_riscv_translate.c.ll:67687]: Duh, right.

**Pull request closed without merging**